messages/eslintrc-incompat.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+/* eslint consistent-return: 0 -- no default case */
+
+module.exports = function({ key }) {
+
+    switch (key) {
+        case "env":
+            return `
+A config object is using the "env" key, which is not supported in flat config system.
+
+Flat config uses "languageOptions.globals" to define global variables for your files.
+
+Please see the following page for information on how to convert your config object into the correct format:
+https://eslint.org/docs/latest/use/configure/migration-guide#configuring-language-options
+            `.trim();
+
+        case "extends":
+            return `
+A config object is using the "extends" key, which is not supported in flat config system.
+
+Instead of "extends", you can include config objects that you'd like to extend from directly in the flat config array.
+
+Please see the following page for more information:
+https://eslint.org/docs/latest/use/configure/configuration-files-new#using-predefined-configurations
```

Comment:
```suggestion
https://eslint.org/docs/latest/use/configure/migration-guide#predefined-configs
```

The section in `configuration-files-new` covers only `eslint:*` configs, while the section in `migration-guide` covers shareable configs as well.